agreement_dynamic/views/agreement.xml

```diff
@@ -0,0 +1,228 @@
+                <field name="section_ids" invisible="1" />
+                <button
+                    type="object"
+                    name="action_view_section"
+                    class="oe_stat_button"
+                    icon="fa-file-text-o"
+                    attrs="{'invisible': [('section_count', '=', 0)]}"
+                    groups="base.group_user"
+                >
+                    <field name="section_count" widget="statinfo" string="Sections" />
+                </button>
+            </xpath>
+            <xpath expr="//field[@name='is_template']" position="after">
+                <field
+                    name="template_id"
+                    attrs="{'readonly': [('signature_date', '!=', False)]}"
+                />
+                <!--
+                editable only for templates
+                 -->
+                <field
+                    name="resource_ref"
+                    attrs="{'readonly': [('is_template', '=', False)]}"
+                />
+                <field
+                    name="condition_domain_global"
+                    widget="domain"
+                    options="{'model': 'agreement', 'in_dialog': True}"
+                    attrs="{'invisible': [('is_template', '=', False)]}"
+                />
+                <button
+                    name="action_wizard_sign_agreement"
+                    string="Lock agreement"
+                    type="object"
+                    class="oe_highlight"
+                    attrs="{'invisible': [('signature_date', '!=', False)]}"
+                    context="{'default_agreement_id': active_id}"
+                />
+            </xpath>
+            <xpath expr="//notebook" position="inside">
+                <page name="dynamic_content" string="Dynamic content">
+                    <!-- Show this when agreement is locked-->
+                    <div attrs="{'invisible': [('signature_date', '=', False)]}">
+                        <p>
+                            <h2>
+                                <strong>LOCKED</strong>
+                            </h2>
+                        </p>
+                    </div>
+                    <group name="content_buttons" col="1">
+                        <button
+                            name="action_validate_content"
+                            string="Validate"
+                            type="object"
+                            class="oe_highlight"
+                            attrs="{'invisible':['|',('resource_ref', '=', False),('signature_date', '!=', False)]}"
+                        />
+                        <button
+                            name="%(agreement_dynamic_document_preview)d"
+                            string="Preview"
+                            type="action"
+                            class="oe_highlight"
+                            attrs="{'invisible':[('resource_ref', '=', False)]}"
+                        />
```

Comment:
What I have just done is to separate model_id from the reference field, and use the reference field as basically a res_id many2one selector. following your instructions from below. So for now, they can select the record to be preview both in templates and agreements. So is this needed?

Reply:
Ok yes, that was the other option, but it needs to be visually clear that it's a preview record and not a permanent record.

Maybe in case of a template, you can prefix the word PREVIEW, then it's fine.